lesson17/src/js/main.js

```diff
@@ -0,0 +1,207 @@
+  const sliderImage = createElementWithClassName("img", "slider__image");
+  sliderImage.src = './images/slide-image01.png'
+  sliderImage.width = 800;
+  sliderImage.height = 800;
+  sliderImage.style.opacity = 1;
+  slideImageArea.appendChild(sliderImage);
+}
+
+function switchPagination(target) {
+  const pagenationCurrent = document.getElementById("js-current");
+  const pagenationTotal = Number(
+    document.getElementById("js-total").textContent
+  );
+  let currentNum = Number(pagenationCurrent.textContent);
+
+  const buttonPrev = target.getAttribute("aria-label") === "previous";
+  const buttonNext = target.getAttribute("aria-label") === "next";
+
+  if (currentNum < pagenationTotal && buttonNext) {
+    pagenationCurrent.textContent = currentNum += 1;
+  } else if (currentNum <= pagenationTotal && buttonPrev) {
+    pagenationCurrent.textContent = currentNum -= 1;
+  }
+}
+
+function switchImg(target) {
+  const active = document.querySelector(".is-active");
+  active.classList.remove("is-active");
+  active[target].classList.add("is-active");
+}
+
+function toggleButtonDisabled() {
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+  const activeImage = document.querySelector(".is-active");
+  const currentIndex = Number(activeImage.dataset.slideIndex);
+  const firstIndex = 1;
+  const lastIndex = slideImages.length;
+  const prevButton = document.getElementById("js-button_previous");
+  const nextButton = document.getElementById("js-button_next");
+  prevButton.disabled = currentIndex === firstIndex;
+  nextButton.disabled = currentIndex === lastIndex;
+}
+
+function callImageData() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(
+        fetchImageData(
+          //https://api.json-generator.com/templates/IZjWl012CAMD/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl : image = 0
+          //https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl&status=503 : 503
+          "https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl"
+        )
+      );
+    }, 3000);
+  });
+}
+
+async function fetchImageData(URL) {
+  const response = await fetch(URL);
+  if (response.ok) {
+    const json = await response.json();
+    return json;
+  } else {
+    console.error(`${response.status}:${response.statusText}`);
+    displayErrorMassage("サーバの通信に失敗しました");
+  }
+}
+
+async function init() {
+  showLoadingImage();
+  renderSlideArea();
+  let slideImageData;
+  try {
+    const json = await callImageData();
+    slideImageData = json.slide;
+    if(slideImageData.length > 0){
+      renderSlideImage(slideImageData);
+      renderPagenation(slideImageData.length);
+      renderSlideButton();
+      toggleButtonDisabled();
+      addSwitchButtonEvent();
+    }else{
+      addNoimage();
+    }
+  } finally {
+    removeLoading();
+  }
+}
```

Comment:
```suggestion
async function init() {
  showLoadingImage();
  renderSlideArea();
  let slideImageData;
  try {
    const json = await callImageData();
    slideImageData = json.slide;
  } finally {
    removeLoading();
  }
  if(slideImageData.length > 0){
    renderSlideImage(slideImageData);
    renderPagenation(slideImageData.length);
    renderSlideButton();
    toggleButtonDisabled();
    addSwitchButtonEvent();
  }else{
    addNoimage();
  }
}
```
Is there a reason why you are running in TRY? If not, you should execute it outside.

Reply:
Thank you for your review!

I've fixed this, so
Thank you.

[1054d6b](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/1054d6bffb166a4603cc3f6d7da6ac3c2c18b301)